### PR TITLE
[expand-placeholder] Add support for multiple-trailing closures

### PR DESCRIPTION
**Pull request by benlangmuir**
Conceptually this depends on https://github.com/apple/swift/pull/31052

---

Since placeholder expansion works with a single placeholder, which is
somewhat at odds with multiple-trailing closures, we eagerly attempt to
expand all consecutive placeholders of closure type. That is, if the API
has multiple closure parameters at the end, expanding any one of them
will transform all of them to the new syntax.

Example

```swift
foo(a: <#T##()->()#>, b: <#T##()->()#>)
```

expanding *either* parameter will produce the following:

```swift
foo {
  <#code#>
} b: {
  <#code#>
}
```

(caveat: the indentation is not part of placeholder expansion, but it's
added here for clarity)

At least for now we do not attempt to corral an existing closure into
the new syntax, so for

```swift
foo(a: { bar() }, b: <#T##()->()#>)
```

The exansion will be

```swift
foo(a: { bar() }) {
  <#code#>
}
```

as it was before.

rdar://59688632

### CODE REVIEW

**Comment by benlangmuir** [comment]
@swift-ci please smoke test

**Comment by benlangmuir** [comment]
@swift-ci please smoke test Linux

File: test/SourceKit/CodeExpand/code-expand-multiple-trailing-closures.swift [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+// CHECK-NEXT: <#code#>
+// CHECK-NEXT: }
+// CHECK-NEXT: b: { () -> Int in
+// CHECK-NEXT: <#code#>
+// CHECK-NEXT: }
+// CHECK-NEXT: }
+
+withTypesAndLabels1(a: <#T##(_ booly: Bool, inty: Int) -> ()#>, b: <#T##(solo: Xyz) -> ()#>)
+// CHECK:      withTypesAndLabels1 {
+// CHECK-NEXT: a: { (booly, inty) in
```
Comment (nathawes) [inline test/SourceKit/CodeExpand/code-expand-multiple-trailing-closures.swift]: This is unrelated to this PR since it's existing behavior, but I feel like most Swift code I've seen doesn't put the parens around the closure params unless they have explicit types. Is there a reason we print them at the moment?

**Review by nathawes — APPROVED** [review]
Looks good to me!

File: test/SourceKit/CodeExpand/code-expand-multiple-trailing-closures.swift (hunk 1 above)
Comment (benlangmuir) [reply test/SourceKit/CodeExpand/code-expand-multiple-trailing-closures.swift]: I don't think there's a good reason, it's just what falls out of the implementation since if you have types or return type you need them.  I agree we should remove them when not needed.

**Comment by benlangmuir** [comment]
Migrated to the main PR #31052
